types/unist/index.d.ts

```diff
@@ -6,13 +6,16 @@
 //                 Titus Wormer <https://github.com/wooorm>
 //                 Junyoung Choi <https://github.com/rokt33r>
 //                 Ben Moon <https://github.com/GuiltyDolphin>
+//                 JounQin <https://github.com/JounQin>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 3.0
 
 /**
  * Syntactic units in unist syntax trees are called nodes.
+ *
+ * @typeParam TData Information from the ecosystem. Useful for more specific `Node#data`.
  */
-export interface Node {
+export interface Node<TData extends object = Data> {
```

Comment:
We could go one step further and make `type` generic as well. The situation is very similar to `data` or `Parent['children'}`.

```ts
export interface Node<TType extends string = string, TData extends object = Data> {
  type: TType;
  data?: TData | undefined;
  position?: Position | undefined;
}
```

I’m not saying we should or shouldn’t. Just saying we could. I think it’s good to discuss this at this point.

Reply:
Yep, similar use case. If we wanted to go _even_ further we could do the same for `Position`, potentially. We should keep an eye on the number of type parameters though - if we add too many it could get confusing.

I'd probably vote for **not** adding this one - it's pretty clear when using `Node`s if you do e.g., `const f = (x: Node<Dat> & { type: 'a' | 'b' })...` seeing as all nodes have types.

Reply:
Personally I won't vote to change like that either, at least in this PR, that should be discussed furtherly.